ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/Divulgence.scala

```diff
@@ -0,0 +1,133 @@
+          assert(event.isCreated, "The transaction should contain a created event")
+
+          val contractId = event.created.get.contractId
+          assert(
+            contractId == divulgence2.contractId,
+            "The only visible event should be the creation of the second contract")
+
+          assert(
+            bobTrees.size == 2,
+            s"Bob should see exactly two transaction trees but sees ${bobTrees.size} instead")
+
+          val createDivulgence2Transaction = bobTrees(0)
+          assert(
+            createDivulgence2Transaction.rootEventIds.size == 1,
+            s"The transaction where Divulgence2 is created should contain exactly one root event contains ${createDivulgence2Transaction.rootEventIds.size} instead"
+          )
+
+          val createDivulgence2 =
+            createDivulgence2Transaction.eventsById(createDivulgence2Transaction.rootEventIds(0))
+          assert(
+            createDivulgence2.kind.isCreated,
+            s"Event expected to be a create"
+          )
+
+          assert(
+            createDivulgence2.getCreated.contractId == divulgence2.contractId,
+            s"The event where Divulgence2 is created should have the same contract identifier as the created contract"
+          )
+
+          val exerciseOnDivulgence2Transaction = bobTrees(1)
+          assert(
+            exerciseOnDivulgence2Transaction.rootEventIds.size == 1,
+            s"The transaction where a choice is exercised on Divulgence2 should contain exactly one root event contains ${exerciseOnDivulgence2Transaction.rootEventIds.size} instead"
+          )
+
+          val exerciseOnDivulgence2 = exerciseOnDivulgence2Transaction.eventsById(
+            exerciseOnDivulgence2Transaction.rootEventIds(0))
+          assert(
+            exerciseOnDivulgence2.kind.isExercised,
+            s"Expected event to be an exercise"
+          )
+
+          assert(exerciseOnDivulgence2.getExercised.contractId == divulgence2.contractId)
+
+          assert(exerciseOnDivulgence2.getExercised.childEventIds.size == 1)
+
+          val exerciseOnDivulgence1 =
+            exerciseOnDivulgence2Transaction.eventsById(
+              exerciseOnDivulgence2.getExercised.childEventIds(0))
+
+          assert(exerciseOnDivulgence1.kind.isExercised)
+
+          assert(exerciseOnDivulgence1.getExercised.contractId == divulgence1.contractId)
+
+          assert(exerciseOnDivulgence1.getExercised.childEventIds.isEmpty)
+
+          assert(
+            aliceTransactions.size == 3,
+            s"Alice should see three transactions but sees ${aliceTransactions.size} instead")
+
+          assert(aliceTransactions.head.events.size == 1)
+
+          assert(aliceTransactions.head.events.head.event.isCreated)
+
+          assert(
+            aliceTransactions.head.events.head.event.created.get.contractId == divulgence1.contractId)
+
+          assert(aliceTransactions.head.events.head.event.created.get.witnessParties == Seq(alice))
```

Comment:
The original test used both parties to request the transactions for this part of the test, to show that bob does not see this create event.
https://github.com/digital-asset/daml/blob/4ff9a6b7eae9e1ae55c5811f7c8c6f4a5a4d25c1/ledger/ledger-api-integration-tests/src/test/itsuite/scala/com/digitalasset/platform/tests/integration/ledger/api/DivulgenceIT.scala#L321

Reply:
Addressed by https://github.com/digital-asset/daml/pull/2282/commits/c7e6538d4efaacb3ed283c726acc0a12dfaaa80f